### PR TITLE
Tests for arm64 kexec, eliminate KexecOpts.Mmap{...} options

### PR DESCRIPTION
**Pull request by hugelgupf**
* Tests for arm64 kexec.
* Instead of allowing users to specify whether to mmap kernel/initramfs, just try the mmap. If it fails, fall back to reading the file.

Depends on https://github.com/u-root/u-root/pull/2885 and https://github.com/u-root/u-root/pull/2886

### CODE REVIEW

File: pkg/boot/linux/load_linux.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		return nil, nil, fmt.Errorf("cannot mmap zero-len file")
+	}
+	d, err := unix.Mmap(int(f.Fd()), 0, int(s.Size()), syscall.PROT_READ, syscall.MAP_PRIVATE)
+	if err != nil {
+		return nil, nil, fmt.Errorf("mmap failed: %w", err)
+	}
+
+	ummap := func() error {
+		if err := unix.Munmap(d); err != nil {
+			return fmt.Errorf("failed to unmap %s: %v", f.Name(), err)
```
Comment (rminnich) [inline pkg/boot/linux/load_linux.go:32]: maybe a %w here?

**Review by rminnich — APPROVED** [review]
just had one comment, your call. 
So the Mmap is gone? IIRC it was added for low memory situations, did that issue go away?

**Comment by hugelgupf** [comment]
> just had one comment, your call. So the Mmap is gone? IIRC it was added for low memory situations, did that issue go away?

The mmap isn't gone -- rather than having to set KexecOptions.MmapKernel we will just always try to mmap the file, and if mmap didn't work, we'll read it. No need to configure it.

**Comment by rminnich** [comment]
the automagic property of this is quite nice ...

File: pkg/boot/linux/load_linux.go (hunk 1 above)
Comment (hugelgupf) [reply pkg/boot/linux/load_linux.go]: done

**Review by rminnich — APPROVED** [review]

**Pull request merged**